### PR TITLE
driver: add a link against swiftCore

### PR DESCRIPTION
**Pull request by compnerd**
This was a missing dependency which results in undefined references.
Add the missing dependency which will also ensure that the build graph
is correct.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @eeckstein

**Comment by compnerd** [comment]
@swift-ci please test

File: tools/driver/CMakeLists.txt [hunk 1]
```diff
@@ -31,6 +31,7 @@ if(${LIBSWIFT_BUILD_MODE} MATCHES "BOOTSTRAPPING.*")
   )
   target_link_libraries(swift-frontend-bootstrapping1
                         PRIVATE
+                          swiftCore-bootstrapping0
```
Comment (eeckstein) [inline tools/driver/CMakeLists.txt:34]: Why is this needed?
`libswift-bootstrapping1` already depends on `swiftCore-bootstrapping0` (https://github.com/apple/swift/blob/669e3f34a6839af68e3bb43aff8d19c5be20eac1/libswift/CMakeLists.txt#L37)

Comment (compnerd) [reply tools/driver/CMakeLists.txt:34]: The users of the library need to specify their link dependencies.  That results in not only the build graph tracking that dependency but actually ensures that the library search path is properly updated to locate the library.  Without the dependency here, we would fail to find the import library for swiftCore resulting in a build failure.

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

**Comment by eeckstein** [comment]
It's better to link the libraries to the SwiftCompilerModules targets. This avoids adding them to all host tools which link the compiler modules. Also, this must be conditional on the bootstrapping mode.
See https://github.com/eeckstein/swift/commit/d2ab30312a971a1b1224302afab5924c3484fda8

**Comment by compnerd** [comment]
CC: @DougGregor 

I think that I am going to close this off for now as I believe that we are investigating a different approach to handling this?
